content/sensu-go/5.5/reference/agent.md

```diff
@@ -112,6 +112,35 @@ HTTP/1.1 202 Accepted
 
 _PRO TIP: You can use the agent API `/events` endpoint to create proxy entities by including a `proxy_entity_name` attribute within the `check` scope._
 
+#### Detecting silent failures
+
+You can use the Sensu agent API in combination with the check time-to-live attribute (TTL) to detect silent failures, creating what's commonly referred to as a "dead man's switch" (source: [Wikipedia][20]).
+By using check TTLs, Sensu is able to set an expectation that a Sensu agent continues to publish events for a check at a regular interval.
+If a Sensu agent fails to publish an event and the check TTL expires, Sensu creates a event with a status of `1` (warning) to indicate the silent failure.
+For more information on check TTLs, see the [the check reference][44].
+
+A great use case for the Sensu agent API is to create a dead man's switch to ensure that backup scripts continue to run successfully at regular intervals.
+If an external source sends a Sensu event with a check TTL to the Sensu agent API, Sensu expects another event from the same external source before the TTL expires.
+
+The following is an example of external event input via the Sensu agent API using a check TTL to create a dead man's switch for MySQL backups. The example uses a check TTL of `25200` seconds (7 hours).
+A MySQL backup script using the following code would be expected to continue to send an event at least once every 7 hours or Sensu creates an alert to indicate the silent failure.
+
+{{< highlight shell >}}
+curl -X POST \
+-H 'Content-Type: application/json' \
+-d '{
+  "check": {
+    "metadata": {
+      "name": "check-mysql-backup-status"
+    },
+    "status": 0,
+    "output": "backed up mysql successfully | size_mb=568",
+    "ttl": 25200
+  }
+}' \
+http://127.0.0.1:3031/events
+{{< /highlight >}}
+
```

Comment:
```suggestion

With this initial event submitted to the Agent API, we have recorded in the backend that our script started, and configured the dead man's switch so that we'll be alerted if the job fails or runs too long. Although it is possible for our script to gracefully handle errors and emit additional monitoring events, this approach allows us to worry less about handling every possible error case, as the lack of additional check result before the 7 hour period elapses will result in an alert.

If our backup script runs successfully, we can send an additional check result without the TTL attribute, which removes the dead man's switch:

{{< highlight shell >}}
curl -X POST \
-H 'Content-Type: application/json' \
-d '{
  "check": {
    "metadata": {
      "name": "mysql-backup-job"
    },
    "status": 0,
    "output": "mysql backup ran successfully!"
  }
}' \
http://127.0.0.1:3031/events
{{< /highlight >}}

By omitting the TTL attribute from this check result, the dead man's switch being monitored by the Sensu backend is also removed, effectively sounding the "all clear" for this iteration of the task.
```